client/core/helpers.go

```diff
@@ -400,6 +400,105 @@ func (ord *OrderReader) MatchReaders() []*matchReader {
 	return readers
 }
 
+// IsMaker returns if the user is the maker in this match.
+func (m *matchReader) IsMaker() bool {
+	return m.Side == order.Maker
+}
+
+// MakerSwapID reutrns the ID of the asset used in the maker's swap.
+func (m *matchReader) MakerSwapID() uint32 {
+	if m.Side == order.Maker {
+		return m.ord.FromID()
+	}
+	return m.ord.ToID()
+}
+
+// TakerSwapID reutrns the ID of the asset used in the taker's swap.
+func (m *matchReader) TakerSwapID() uint32 {
+	if m.Side == order.Maker {
+		return m.ord.ToID()
+	}
+	return m.ord.FromID()
+}
+
+// MakerSwapSymbol reutrns the symbol of the asset used in the maker's swap.
+func (m *matchReader) MakerSwapSymbol() string {
+	if m.Side == order.Maker {
+		return m.ord.FromSymbol()
+	}
+	return m.ord.ToSymbol()
+}
+
+// TakerSwapSymbol reutrns the symbol of the asset used in the taker's swap.
+func (m *matchReader) TakerSwapSymbol() string {
+	if m.Side == order.Maker {
+		return m.ord.ToSymbol()
+	}
+	return m.ord.FromSymbol()
+}
+
+// MakerSwap returns the coin created by the maker's swap transaction.
+func (m *matchReader) MakerSwap() *Coin {
+	if m.Side == order.Maker {
+		return m.Swap
+	}
+	return m.CounterSwap
+}
+
+// TakerSwap returns the coin created by the taker's swap transaction.
+func (m *matchReader) TakerSwap() *Coin {
+	if m.Side == order.Maker {
+		return m.CounterSwap
+	}
+	return m.Swap
+}
+
+// MakerRedeem returns the coin created by the maker's redeem transaction.
+func (m *matchReader) MakerRedeem() *Coin {
+	if m.Side == order.Maker {
+		return m.Redeem
+	}
+	return m.CounterRedeem
+}
+
+// TakerRedeem returns the coin created by the taker's redeem transaction.
+func (m *matchReader) TakerRedeem() *Coin {
+	if m.Side == order.Maker {
+		return m.CounterRedeem
+	}
+	return m.Redeem
+}
+
+// ShowMakerSwap returns whether or not to display the maker swap section
+// on the match card.
+func (m *matchReader) ShowMakerSwap() bool {
+	return m.MakerSwap() != nil || !m.Revoked
+}
```

Comment:
I think should also not show all of these for cancel orders

![image](https://user-images.githubusercontent.com/34998433/187620669-4e2c3f4f-3734-4383-a16a-c016604bbbcf.png)